code/game/machinery/autolathe.dm

```diff
@@ -77,6 +79,12 @@
 	materials.retrieve_all()
 	return ..()
 
+/obj/machinery/autolathe/proc/on_tool_attack(datum/source, atom/tool, mob/user)
+	// Allows screwdrivers to be recycled on harm intent
+	if(istype(tool, /obj/item/screwdriver) && user.a_intent == INTENT_HARM)
```

Comment:
Please check that tool is an item, then check that its tool_behavior is screwdriver instead, we're trying to get away from direct istype checks for tools. Something like this.
`I.tool_behaviour == TOOL_SCREWDRIVER`

Reply:
Ah, that makes a lot of sense. Will do.